Form/FormFlow.php

```diff
@@ -722,7 +754,16 @@ protected function applyDataFromSavedSteps() {
 	 * {@inheritDoc}
 	 */
 	public function createForm() {
-		return $this->createFormForStep($this->currentStepNumber);
+		$form = $this->createFormForStep($this->currentStepNumber);
+
+		if ($this->expired) {
+			if ($this->hasListeners(FormFlowEvents::FLOW_EXPIRED)) {
+				$event = new FlowExpiredEvent($this, $form);
+				$this->eventDispatcher->dispatch(FormFlowEvents::FLOW_EXPIRED, $event);
+			}
+		}
```

Comment:
I think my thought process was that checking for the entire flow being expired is a flow-level function rather than creating a form for a specific step, and so it made sense to put it at the "createForm" level rather than "createFormForStep".